src/stream.cpp

```diff
@@ -18,14 +18,16 @@ extern "C" {
 }
 
 #include "config.h"
+#include "entry_handler.h"
```

Comment:
What entry handler functions are these (stream.cpp, rtsp.cpp, etc) using?

Reply:
Let me double check... it might have been from before I created the `globals`.

Reply:
Here's a table of the files I added `entry_handler` to. I'll remove it from the ones without a check.

| source file                    | mail               | nv_prefs_instance  | lifetime           | service_ctlr       | launch_ui          | launch_ui_with_path |
|--------------------------------|--------------------|--------------------|--------------------|--------------------|--------------------|---------------------|
| src/audio.cpp                  | :white_check_mark: |                    |                    |                    |                    |                     |
| src/config.cpp                 |                    |                    |                    | :white_check_mark: | :white_check_mark: |                     |
| src/confighttp.cpp             | :white_check_mark: |                    |                    |                    |                    |                     |
| src/input.cpp                  |                    |                    |                    |                    |                    |                     |
| src/nvhttp.cpp                 | :white_check_mark: |                    |                    |                    |                    |                     |
| src/platform/linux/audio.cpp   |                    |                    |                    |                    |                    |                     |
| src/platform/linux/misc.cpp    |                    |                    | :white_check_mark: |                    |                    |                     |
| src/platform/linux/x11grab.cpp |                    |                    |                    |                    |                    |                     |
| src/platform/macos/misc.mm     |                    |                    | :white_check_mark: |                    |                    |                     |
| src/platform/windows/input.cpp |                    |                    |                    |                    |                    |                     |
| src/platform/windows/misc.cpp  |                    | :white_check_mark: | :white_check_mark: |                    |                    |                     |
| src/rtsp.cpp                   | :white_check_mark: |                    |                    |                    |                    |                     |
| src/stream.cpp                 |                    |                    |                    |                    |                    |                     |
| src/system_tray.cpp            |                    |                    | :white_check_mark: |                    | :white_check_mark: | :white_check_mark:  |
| src/upnp.cpp                   | :white_check_mark: |                    |                    |                    |                    |                     |
| src/video.cpp                  |                    |                    |                    |                    |                    |                     |

`src/platform/linux/audio.cpp` gives the following errors without `entry_handler.h`:

- No type named 'thread' in namespace 'std'; did you mean 'boost::thread'?: 201
- No member named 'thread' in namespace 'std': 238
- Expected ';' after expression: 238
- Expected ';' after expression: 239

Probably a better fix than inclduing `entry_handler.h` exists.

Reply:
Solved audio.cpp by including `<thread>`

Reply:
I think moving the whole `mail` namespace into `globals.cpp/h` would clean things up nicely. It's a process-wide communication mechanism, so I think it makes sense for it to live with other global state.

Reply:
Sounds good. I'll work on that.

Reply:
Done in https://github.com/LizardByte/Sunshine/pull/2127/commits/a71d3b68ab325989f3ab3ecab9b74f280d8b2c62

Reply:
I made a few more tweaks, but I think this is ready now.

- moved `nvprefs_instance` to `globals`
- added/updated some documentation blocks in `globals.cpp` and `entry_handler.cpp`
- changed the input validation of the `creds` arg to the just use the `help` function within the `args` namespace